src/layout/List/ListComponent.tsx

```diff
@@ -1,133 +1,172 @@
+      return result;
+    }) ?? [];
+
+  const selectedRow = filteredRows.find((row) => Object.keys(formData).every((key) => row[key] === formData[key]));
+
+  function handleRowSelect({ selectedValue }: ChangeProps<Row>) {
+    const next: Row = {};
     for (const binding of Object.keys(bindings)) {
       next[binding] = selectedValue[binding];
     }
     setValues(next);
-  };
-
-  const tableHeadersValues = { ...tableHeaders };
-  for (const key in tableHeaders) {
-    tableHeadersValues[key] = langAsString(tableHeaders[key]);
   }
 
-  const selectedRow: Record<string, string> = React.useMemo(() => {
-    let matchRow: boolean[] = [];
-    if (!formData || Object.keys(formData).length === 0) {
-      return {};
-    }
-    for (const row of calculatedDataList) {
-      for (const key in formData) {
-        matchRow.push(formData[key] === row[key]);
-      }
-      if (!matchRow.includes(false)) {
-        return row;
-      }
-      matchRow = [];
-    }
-    return {};
-  }, [formData, calculatedDataList]);
-
-  const renderPagination = () =>
-    pagination && (
-      <Pagination
-        numberOfRows={data?._metaData.totaltItemsCount ?? 0}
-        rowsPerPageOptions={pagination?.alternatives ? pagination?.alternatives : []}
-        rowsPerPage={pageSize}
-        onRowsPerPageChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
-          setPageSize(parseInt(event.target.value, 10));
-        }}
-        currentPage={pageNumber}
-        setCurrentPage={(newPage: number) => {
-          setPageNumber(newPage);
-        }}
-        descriptionTexts={((language && language['list_component']) || {}) as unknown as DescriptionText}
-      />
-    );
+  function shouldIncludeColumn(key: string): boolean {
+    return !isMobile || !tableHeadersMobile || tableHeadersMobile.includes(key);
+  }
 
-  const config: LegacyResponsiveTableConfig<Record<string, string>> = {
-    rows: calculatedDataList,
-    headers: tableHeadersValues,
-    showColumnsMobile: tableHeadersMobile,
-    columnSort: {
-      onSortChange: ({ column, next }) => {
-        setSortColumn(column);
-        setSortDirection(next);
-      },
-      sortable: sortableColumns ? sortableColumns : [],
-      currentlySortedColumn: sortColumn,
-      currentDirection: sortDirection,
-    },
-    rowSelection: {
-      onSelectionChange: (row) => handleChange({ selectedValue: row }),
-      selectedValue: selectedRow,
-    },
-    renderCell: Object.keys(tableHeaders).reduce(
-      // Add lang as the renderCell function for all inputs that are of type string.
-      (acc, next) => ({ ...acc, [next]: (v) => (typeof v === 'string' ? lang(v) : v) }),
-      {},
-    ),
-    footer: renderPagination(),
-  };
+  function isRowSelected(row: Row): boolean {
+    return JSON.stringify(selectedRow) === JSON.stringify(row);
+  }
 
   return (
     <ComponentStructureWrapper
       node={node}
       label={{ node, renderLabelAs: 'legend' }}
```

Comment:
Now that we are able to, i would prefer using a `<caption />` instead of a legend for the tables title in order to get an accessible name for the table  for screen-readers out of the box. I believe we have an implementation of a Caption component that uses the designsystem Label under `src/components/form/Caption.tsx`